### ISSUE TITLE
Cockpit on CentOS 8 fails to restart

### ISSUE THREAD

**Issue by VinzenzBildstein** [cockpit-project/cockpit#15121]
Cockpit version: cockpit-224.2-1.el8.x86_64
OS: CentOS 8.3

I had problems accessing cockpit so I tried restarting the service. This would always show two lines while deactivating:
``` 
systemd[1]: cockpit.service: State 'stop-sigterm' timed out. Killing.
systemd[1]: cockpit.service: Processes still around after SIGKILL. Ignoring.
```

I looked a bit further into this and found that there are a bunch of processes still hanging around:
```
[root@host ~]# ps -fp `pgrep kpit`
UID          PID    PPID  C STIME TTY      STAT   TIME CMD
root      758130       1  0 Jan08 ?        Ds     0:00 (kpit-tls)
root      962652       1  0 Jan08 ?        Ds     0:00 (kpit-tls)
root     2372538       1  0 11:55 ?        Ds     0:00 (kpit-tls)
root     2394185       1  0 12:08 ?        Ds     0:00 (kpit-tls)
```

I tried to see if restarting the socket would help getting rid of these processes, but that fails. ```journalctl -u cockpit.socket``` yields
```
Jan 11 12:20:07 host systemd[1]: cockpit.socket: Failed to listen on sockets: Address already in use
Jan 11 12:20:07 host systemd[1]: Starting Cockpit Web Service Socket.
Jan 11 12:20:07 host systemd[1]: cockpit.socket: Failed with result 'resources'.
Jan 11 12:20:07 host systemd[1]: Failed to listen on Cockpit Web Service Socket.
```

And checking with ```ss -tulpen | grep 9090``` shows
```
tcp    LISTEN  82       128                      *:9090                 *:*      users:(("(kpit-tls",pid=2394185,fd=71),("(kpit-tls",pid=2372538,fd=71),("(kpit-tls",pid=962652,fd=40),("(kpit-tls",pid=758130,fd=40)) ino:8685894 sk:30 v6only:0 <->
```

I can't kill these processes as they are all in uninterruptible sleep. I don't want to just restart the server without understanding what caused this situation.

**Comment by martinpitt** [comment]
The "D" state (uninterruptible kernel sleep) is unfortunately rather grim -- I am afraid I don't know how to properly debug these, as tools like ptrace (strace, gdb), and even sending signals don't work. This smells like a kernel or hardware bug, I'm afraid. Do you see anything relevant in `journalctl` or `dmesg` around that time?

(FTR, we run CI on CentOS 8, and have lots of users on CentOS, it is not generally broken nor reproducible)

**Comment by VinzenzBildstein** [comment]
I tried to find when this happened, but it is hard to pin point. So far I haven't found anything in the logs. dmesg definitely doesn't seem to reach back far enough (it seems the firewall sends a bunch of messages to the kernel ring buffer ...). 

It has run fine on the CentOS 8 server for a while, so I'm not surprised that a lot of people have no problem running it on CentOS. I just posted it here on the off chance that someone else has run across the same issue and maybe had a solution. If I ever find out what caused this I will update.
